packages/wdio-browserstack-service/src/Percy/Percy-Handler.ts

```diff
@@ -0,0 +1,176 @@
+                await (this._isAppAutomate ? PercySDK.screenshotApp((this._browser.percyCaptureMap as PercyCaptureMap).getName( sessionName ? sessionName : (this.sessionName as string), eventName)) : await PercySDK.screenshot(this._browser, (this._browser.percyCaptureMap as PercyCaptureMap).getName( sessionName ? sessionName : (this.sessionName as string), eventName)))
+                this._percyScreenshotCounter -= 1
+            }
+        } catch (err: any) {
+            this._percyScreenshotCounter -= 1;
+            (this._browser.percyCaptureMap as PercyCaptureMap).decrement(sessionName ? sessionName : (this.sessionName as string), eventName as string)
+            PercyLogger.error(`Error while trying to auto capture Percy screenshot ${err}`)
+        }
+    }
+
+    async before () {
+        this._browser.percyCaptureMap = new PercyCaptureMap()
+    }
+
+    deferCapture(sessionName: string, eventName: string | null) {
+        /* Service doesn't wait for handling of browser commands so the below counter is used in teardown method to delay service exit */
+        this._percyScreenshotCounter += 1
+        this.percyDeferredScreenshots.push({ sessionName, eventName })
+    }
+
+    isDOMChangingCommand(args: BeforeCommandArgs): boolean {
+        if ((args.method as string) === 'POST') {
+            if (PERCY_DOM_CHANGING_COMMANDS_ENDPOINTS.includes((args.endpoint as string))) {
+                return true
+            } else if ((args.endpoint as string).includes('/session/:sessionId/element') && (args.endpoint as string).includes('click')) {
+                /* click element */
+                return true
+            } else if ((args.endpoint as string).includes('/session/:sessionId/element') && (args.endpoint as string).includes('clear')) {
+                /* clear element */
+                return true
+            } else if ((args.endpoint as string).includes('/session/:sessionId/execute') && args.body?.script) {
+                /* execute script sync / async */
+                return true
+            } else if ((args.endpoint as string).includes('/session/:sessionId/touch')) {
+                /* Touch action for Appium */
+                return true
+            }
+        } else if ((args.method as string) === 'DELETE' && (args.endpoint as string) === '/session/:sessionId') {
+            return true
+        }
+        return false
+    }
+
+    async cleanupDeferredScreenshots() {
+        this.isPercyCleanupProcessingUnderway = true
+        for await (const entry of this.percyDeferredScreenshots) {
+            await this.percyAutoCapture(entry.eventName, entry.sessionName)
+        }
+        this.percyDeferredScreenshots = []
+        this.isPercyCleanupProcessingUnderway = false
+    }
+
+    async #sleep(ms: number) {
+        return new Promise(resolve => setTimeout(resolve, ms))
+    }
```

Comment:
This can be a utility function, not a class method.

Reply:
Done